pkg/kdump/config-client.js

```diff
@@ -113,21 +124,185 @@ export class ConfigFile {
+                if (!key || key in knownKeys || key in deprecatedKeys)
+                    return;
+                // if we have a UUID, LABEL or /dev in the value, we can be pretty sure it's a mount option
+                const value = JSON.stringify(this.settings._internal[key]).toLowerCase();
+                if (value.indexOf("uuid") > -1 || value.indexOf("label") > -1 || value.indexOf("/dev") > -1) {
+                    this.settings.targets.mount = {
+                        type: "mount",
+                        path: path.value,
+                        fsType: key,
+                        partition: this.settings._internal[key].value,
+                    };
+                } else {
+                    // TODO: check for know filesystem types here
+                }
+            }
+        });
+
+        // default to local if no target configured
+        if (Object.keys(this.settings.targets).length === 0)
+            this.settings.targets.local = { type: "local", path: path.value };
+
+        // only allow compression if there is no core collector set or it's set to makedumpfile
+        this.settings.compression.allowed = (
+            !("core_collector" in this.settings._internal) ||
+            (this.settings._internal.core_collector.value.trim().indexOf("makedumpfile") === 0)
+        );
+        // compression is enabled if we have a core_collector command with the "-c" parameter
+        this.settings.compression.enabled = (
+            ("core_collector" in this.settings._internal) &&
+            this.settings._internal.core_collector.value &&
+            (this.settings._internal.core_collector.value.split(" ").indexOf("-c") != -1)
+        );
+    }
+
+    /* update single _internal setting to given value
+     * make sure setting exists if value is not empty
+     */
+    _updateSetting(settings, key, value) {
+        if (key in settings._internal) {
+            if (value)
+                settings._internal[key].value = value;
+            else
+                delete settings._internal[key];
+        } else {
+            if (value)
+                settings._internal[key] = { value: value };
+        }
+    }
+
+    /* transform settings from model back to _internal format
+     * this.settings = current state from file
+     * settings = in-memory state from UI
+     */
+    _persistSettings(settings) {
+        // target
+        if (Object.keys(settings.targets).length > 0) {
+            const target = Object.values(settings.targets)[0];
+            this._updateSetting(settings, "path", target.path);
+
+            // wipe old target settings
+            for (const key in this.settings.targets) {
+                const oldTarget = this.settings.targets[key];
+                if (oldTarget.type == "mount") {
+                    delete settings._internal[oldTarget.fsType];
+                } else if (oldTarget.type == "ssh") {
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17544-20220719-143653-7a0d5780-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)